wordpress-mobile/WordPress-Android
# Update .configure to new format

I am setting up my M1 to run the Android project and unfortunately getting ruby toolchain to work is very troublesome. So, I've decided to directly use the https://github.com/Automattic/configure project to apply the configuration. I've had a few issues with it, but (with @jkmassel) we realized that this was because `"project_name": "WordPress-Android",` uses the old key format and `"project_name": "wordpress-android",` uses the new one which is what the configure project uses. Long story short, here is what this PR does:

1. Update the `project_name` field in `.configure` file to `wordpress-android`
2. Run `configure update` which puts the files in their new location.

I've not tested the ruby toolchain with these changes, but if there is anything wrong with it Installable build should fail 🤞 

**To test:**
* Probably green CI would be enough

For the extra cautious:
* Install the created installable apk and make sure it runs
* Run `bundle exec fastlane configure apply` and make sure it succeeds

## Regression Notes
1. Potential unintended areas of impact
* Build time failure for installable or release builds

2. What I did to test those areas of impact (or what existing automated tests I relied on)
* Installable build CI task should test it

3. What automated tests I added (or what prevented me from doing so)
N/A

PR submission checklist:
